### PR TITLE
Split code / Use modules

### PR DESCRIPTION
**Pull request by SoyOscarRH**
This PR is just moving code into different file (ECMAScript modules)
The code is mainly the same, logic is intact, all features and bugs are still here

I want to just this PR so I can create more "modular" focusing on one feature]
Here is a demo of our current state:
ezgif com-video-to-gif

### CODE REVIEW

File: fudge-ratio/src/main/webapp/calculator/main.js [hunk 1]
```diff
@@ -1,80 +1,27 @@
+import { saveSessionState, recoverSessionState } from "../code/save.js";
```
Comment (yaaowusu00) [inline fudge-ratio/src/main/webapp/calculator/main.js:1]: I wasn't familiar with export and import before this so I just looked it up. Thanks for introducing something new

File: fudge-ratio/src/main/webapp/code/time.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      this.seconds = 0;
+    }
+    if (this.minutes == 60) {
+      this.hours = (this.hours + 1) % 24;
+      this.minutes = 0;
+    }
+  }
+
+  getText() {
+    return this.padd(this.hours) + ":" + this.padd(this.minutes) + ":" + this.padd(this.seconds);
```
Comment (yaaowusu00) [inline fudge-ratio/src/main/webapp/code/time.js:39]: did you add this method or did we have it before?

**Review by yaaowusu00 — APPROVED** [review]
The way you split up the files looks good to me. Thanks for going through and making sure everything still works

File: fudge-ratio/src/main/webapp/calculator/main.js (hunk 1 above)
Comment (SoyOscarRH) [reply fudge-ratio/src/main/webapp/calculator/main.js:1]: Thanks 😊, modules have a large history in Javascript, the thing is that they had just been standardized in the last ~3 years (ECMAscript modules)

Here is a great link to learn more:
https://eloquentjavascript.net/10_modules.html#h_hF2FmOVxw7

File: fudge-ratio/src/main/webapp/code/time.js (hunk 2 above)
Comment (SoyOscarRH) [reply fudge-ratio/src/main/webapp/code/time.js:39]: No, we did not, I add it, tried to add as little as I could, but great eye

**Pull request merged**